### DISCUSSION TITLE
Add buf(validation) support for protobuf

### DISCUSSION THREAD

**Discussion by AlmogAlmado** [redpanda-data/connect#2325] [Ideas]
Is there a plan to add Buf? 
This can be very helpful because it gives the capability to read protobuf schema with validation

https://buf.build/
https://buf.build/docs/installation
*We just need to install it on the image
---------------
Scheme example:
syntax = "proto3";

package etl;

import "buf/validate/validate.proto";   (currently break)
import "google/protobuf/struct.proto";
import "google/protobuf/wrappers.proto";

**Comment by mihaitodor** [comment]
Hey @AlmogAlmado, I'm not sure what the ask is. Are you looking to have the `buf/validate/validate.proto` file and its dependencies inside the Benthos Docker image? If yes, I don't think that should be part of the main distribution. Volume mounting a folder with those `.proto` files or creating a custom Docker image should do the trick in this case.

**Reply by AlmogAlmado** [reply]
@mihaitodor, thanks for the quick assistance as usual. I found a workaround similar to your suggestion. 
I was just wondering if there are any plans to implement it regardless. 
Thanks.

**Reply by mihaitodor** [reply]
You’re welcome! I’m not sure what you need to have implemented. If you’re looking to have protobuf libraries inside the official releases of the Benthos Docker image, I don’t think that really makes sense.
